pkg/systemd/hwinfo.jsx

```diff
@@ -154,25 +154,20 @@ function availableMitigations() {
     });
 }
 
-class CPUSecurityMitigationsDialog extends React.Component {
-    static contextType = DialogsContext;
+const CPUSecurityMitigationsDialog = () => {
+    const [nosmt, setNoSMT] = React.useState(undefined);
+    const [alert, setAlert] = React.useState(undefined);
+    const [rebooting, setRebooting] = React.useState(false);
 
-    constructor(props) {
-        super(props);
-        this.saveAndReboot = this.saveAndReboot.bind(this);
-        this.state = {
-            nosmt: undefined,
-            alert: undefined,
-            rebooting: false,
-        };
+    React.useEffect(() => {
```

Comment:
I'd rather we use `useInit` for that -- it's more abstract and better communicates what it does. useEffect() can be used for lots of other things.